classes/Cart.php

```diff
@@ -1819,9 +1814,9 @@ public function getPackageList($flush = false)
 			$id_warehouse = 0;
 			foreach ($warehouse_count_by_address[$product['id_address_delivery']] as $id_war => $val)
 			{
-				if (in_array((int)$id_war, $product['warehouse_list']))
+				if (array_key_exists((int)$id_war, $product['warehouse_list']))
 				{
-					$product['carrier_list'] = array_merge($product['carrier_list'], Carrier::getAvailableCarrierList(new Product($product['id_product']), $id_war, $product['id_address_delivery'], null, $this));
+					$product['carrier_list'] = array_replace($product['carrier_list'], Carrier::getAvailableCarrierList(new Product($product['id_product']), $id_war, $product['id_address_delivery'], null, $this));
```

Comment:
array_replace is php 5.3, we still assume compatibility with php 5.2, soon we will remove it, but for this moment...